### PR TITLE
add support for Meta Programming System (MPS)

### PR DESCRIPTION
**Pull request by coolya**
Adding support for Meta Programming System (MPS).
Meta Programming System is a open source tool from JetBrains:

https://www.jetbrains.com/mps/

## Description
It is used to define custom (programming) languages and code generators
and it also used to use these languages. MPS stores the programs in its
AST form serialised to XML. MPS by default uses two file extensions:
most commonly .mps:

https://github.com/search?utf8=✓&q=extension%3Amps+persistence+&type=Code

and far less common .mpsr

https://github.com/search?utf8=✓&q=extension%3Ampsr+persistence+&type=Code

The later is used when programms are stored over mutiple files the first
is used when a complete programm is stored in a single file.

## Checklist:

- [ ] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [x ] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=✓&q=extension%3Amps+persistence+&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - a [file](https://github.com/coolya/mps-generator-editors/blob/master/solutions/de.itemis.mps.generators.editors.build.meta/models/all/allScripts.mpsr) describing how a language in MPS should be build
      - a [file](https://github.com/coolya/mps-generator-editors/blob/master/languages/de.itemis.mps.generator.editors/models/editor.mps) describing the editor aspect of a language
    - Sample license(s): Apache 2.0 https://github.com/coolya/mps-generator-editors/blob/master/LICENSE
  - [ ] I have included a syntax highlighting grammar.
    - I did not do that since the files are xml files where a grammar already existed.
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.
    - I did not do that as there where no other languages associated with this extension. 

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/
  - New: https://github-lightshow.herokuapp.com/

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -2753,6 +2753,17 @@ Meson:
   tm_scope: source.meson
   ace_mode: text
   language_id: 799141244
+Meta Programming System:
+  type: programming
+  color: "#007800"
+  extensions:
+  - ".mps"
+  - ".mpsr"
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Given [the search request you provided](https://github.com/search?utf8=%E2%9C%93&q=extension%3Ampsr+persistence+&type=Code), I don't think there are enough `.mpsr` files to mandate its support in Linguist yet.

**Comment by pchaigno** [comment]
Flagging as stale.

**Comment by ice1000** [comment]
There's now 329 results, larger than 100!
@pchaigno Could you please re-open this issue?

**Comment by ice1000** [comment]
@coolya Hi! Are you interested in addressing the comments about adding an extension to XML?

**Comment by coolya** [comment]
Sorry for the huge delay. I some missed this back in April then entirely forgot about it. Sorry for that. 

Regarding the change to XML I'm not so sure if that aligns with my intention behind this PR. My motivation behind this change was to help people to understand what the content of the repository is. I think indexing the files as XML wouldn't help there.

I would argue files from MPS are more like e.g. SVG files. While SVG files are technically XML, a user  wouldn't typically look at them in their textual/xml representation. Of course looking at these files in a text editor works and if they are small enough it's possible to make sense of the content but usually this is not what people tend to do. For MPS files this is similar, they are technically XML but in all typical use cases these files are never read by human nor edited directly.

If I take a look at the other files classified as XML, they are typically configuration files which don't make up a large part of a codebase, or XML files that also edited as XML with a text editor. In a project that uses MPS the MPS files usually take up a significant part of the code in the repository.  On the other hand ANT files are classified as dedicated language even if these files are typically edited as XML. I'm a bit puzzled how the decision is taken if it's a dedicated language or not. 

I know the community around MPS is small compared to other programming languages or tools and maybe it's no relevant enough to show it as a first class language on Github and I can totally understand that.

**Pull request closed without merging**